### PR TITLE
ADRECP-74 | Add integration test and impl for contact preferences back end connector

### PR DESCRIPTION
**Pull request by ajchilts**
…k end connector

### CODE REVIEW

File: it/test/connectors/WireMockHelper.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * limitations under the License.
+ */
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+
+trait WireMockHelper extends BeforeAndAfterAll with BeforeAndAfterEach {
+  this: Suite =>
+  protected val server: WireMockServer = new WireMockServer(wireMockConfig().dynamicPort())
```
Comment (simondrugan16) [inline it/test/connectors/WireMockHelper.scala:41]: Does .dynamicPort sort all the config out automatically? If not you need to override your config here and set your outbound ports to the same one the WireMockServer is running on

Comment (simondrugan16) [reply it/test/connectors/WireMockHelper.scala:41]: Ah just noticed you were doing this in the spec itself so ignore this

Comment (ajchilts) [reply it/test/connectors/WireMockHelper.scala:41]: @simondrugan16  - I based this on your returns FE. 

The port is overridden at the top of the connector spec. 

Screenshot 2024-11-21 at 11 24 49

Comment (ajchilts) [reply it/test/connectors/WireMockHelper.scala:41]: @simondrugan16  -  I am keen to use the same or similar code / test style in the new services such that they are easy to follow for new dev's getting to grips with the entire suite.

**Pull request merged**